Signal/src/Jobs/MessageFetcherJob.swift

```diff
@@ -113,56 +118,57 @@ public class MessageFetcherJob: NSObject {
         )
     }
 
-    private func buildEnvelope(messageDict: [String: Any]) -> OWSSignalServiceProtosEnvelope? {
-        let builder = OWSSignalServiceProtosEnvelopeBuilder()
+    private func buildEnvelope(messageDict: [String: Any]) -> SSKEnvelope? {
 
         guard let typeInt = messageDict["type"] as? Int32 else {
             Logger.error("\(self.logTag) message body didn't have type")
             return nil
         }
 
-        guard let type = OWSSignalServiceProtosEnvelopeType(rawValue: typeInt) else {
+        guard let type: SSKEnvelope.SSKEnvelopeType = SSKEnvelope.SSKEnvelopeType(rawValue: typeInt) else {
             Logger.error("\(self.logTag) message body type was invalid")
             return nil
         }
-        builder.setType(type)
 
         guard let timestamp = messageDict["timestamp"] as? UInt64 else {
             Logger.error("\(self.logTag) message body didn't have timestamp")
             return nil
         }
-        builder.setTimestamp(timestamp)
 
         guard let source = messageDict["source"] as? String else {
             Logger.error("\(self.logTag) message body didn't have source")
             return nil
         }
-        builder.setSource(source)
 
         guard let sourceDevice = messageDict["sourceDevice"] as? UInt32 else {
             Logger.error("\(self.logTag) message body didn't have sourceDevice")
             return nil
         }
-        builder.setSourceDevice(sourceDevice)
 
-        if let encodedLegacyMessage = messageDict["message"] as? String {
-            Logger.debug("\(self.logTag) message body had legacyMessage")
-            if let legacyMessage = Data(base64Encoded: encodedLegacyMessage) {
-                builder.setLegacyMessage(legacyMessage)
+        let legacyMessage: Data? = {
+            if let encodedLegacyMessage = messageDict["message"] as? String {
+                Logger.debug("\(self.logTag) message body had legacyMessage")
+                if let legacyMessage = Data(base64Encoded: encodedLegacyMessage) {
+                    return legacyMessage
+                }
             }
-        }
+            return nil
+        }()
 
-        if let encodedContent = messageDict["content"] as? String {
-            Logger.debug("\(self.logTag) message body had content")
-            if let content = Data(base64Encoded: encodedContent) {
-                builder.setContent(content)
+        let content: Data? = {
+            if let encodedContent = messageDict["content"] as? String {
+                Logger.debug("\(self.logTag) message body had content")
+                if let content = Data(base64Encoded: encodedContent) {
+                    return content
+                }
             }
-        }
+            return nil
+        }()
 
-        return builder.build()
+        return SSKEnvelope(timestamp: timestamp, source: source, sourceDevice: sourceDevice, type: type, content: content, legacyMessage: legacyMessage)
```

Comment:
It's a shame to lose the builder pattern.  It's so nice when there are optional or nested protos.

Reply:
On the other hand, this ensures we're setting all required params.

If we wanted something more like the builder pattern, we could make the properties `var` instead of `let`.